src/realm/object-store/c_api/sync.cpp

```diff
@@ -646,34 +655,70 @@ realm_sync_subscription_set_insert_or_assign_query(realm_flx_sync_mutable_subscr
     });
 }
 
+RLM_API bool realm_sync_subscription_set_erase_by_id(realm_flx_sync_mutable_subscription_set_t* subscription_set,
+                                                     const realm_object_id_t* id, bool* erased)
+{
+    REALM_ASSERT(subscription_set != nullptr && id != nullptr);
+    *erased = false;
+    return wrap_err([&] {
+        auto it = std::find_if(subscription_set->begin(), subscription_set->end(), [id](const Subscription& sub) {
+            return from_capi(*id) == sub.id();
+        });
+        if (it == subscription_set->end())
+            return false;
+        subscription_set->erase(it);
+        *erased = true;
+        return true;
+    });
+}
+
 RLM_API bool realm_sync_subscription_set_erase_by_name(realm_flx_sync_mutable_subscription_set_t* subscription_set,
-                                                       const char* name) noexcept
+                                                       const char* name, bool* erased)
 {
     REALM_ASSERT(subscription_set != nullptr && name != nullptr);
+    *erased = false;
     return wrap_err([&]() {
         if (auto it = subscription_set->find(name); it != subscription_set->end()) {
             subscription_set->erase(it);
+            *erased = true;
             return true;
         }
         return false;
     });
 }
 
 RLM_API bool realm_sync_subscription_set_erase_by_query(realm_flx_sync_mutable_subscription_set_t* subscription_set,
-                                                        realm_query_t* query) noexcept
+                                                        realm_query_t* query, bool* erased)
 {
     REALM_ASSERT(subscription_set != nullptr && query != nullptr);
+    *erased = false;
     return wrap_err([&]() {
         if (auto it = subscription_set->find(query->get_query()); it != subscription_set->end()) {
             subscription_set->erase(it);
+            *erased = true;
+            return true;
+        }
+        return false;
+    });
+}
+
+RLM_API bool realm_sync_subscription_set_erase_by_results(realm_flx_sync_mutable_subscription_set_t* subscription_set,
+                                                          realm_results_t* results, bool* erased)
+{
+    REALM_ASSERT(subscription_set != nullptr && results != nullptr);
+    *erased = false;
+    return wrap_err([&]() {
+        if (auto it = subscription_set->find(results->get_query()); it != subscription_set->end()) {
+            subscription_set->erase(it);
+            *erased = true;
             return true;
         }
         return false;
```

Comment:
Should return `true` here as well. Return value only indicates error or not, returning false is conceptually the same as throwing an exceptiopn, ie. it signals an error.

Reply:
This problem occurs elsewhere as well

Reply:
ok, let me change it a little bit